src/routes/announcement/[slug]/+page.server.js

```diff
@@ -14,7 +14,8 @@ export async function load({ params }) {
   /** @type {import('$lib/server/post').PostRecord} */
   let post
   try {
-    post = await getPostBySlug(slug)
+    post = await Post.getPostBySlug(slug)
```

Comment:
Since we have a need for retrieving the content of a specific Post, whether it's an Announcement or a BlogPost, and depending on the specific use-case here, either add `getAnnouncementBySlug(slug:string)` and `getBlogpostBySlug(slug:string)` to their appropriate classes, or keep `/lib/server/post.js` for this specific use-case. 

Personally, I think it's cleaner (but a little more typing or copy-paste) to add to the two classes rather than keeping a Post class around for just this purpose. YMMV ("Your mileage may vary" which I use to denote something left up to personal preference.)